xdsl/backend/riscv/lowering/convert_memref_to_riscv.py

```diff
@@ -263,41 +266,14 @@ def match_and_rewrite(self, op: memref.GlobalOp, rewriter: PatternRewriter):
                 f"Unsupported memref.global initial value: {initial_value}"
             )
 
-        memref_type = op.type
-        element_type = memref_type.element_type
-
-        # Only handle a small subset of elements
-        # Might be useful as a helper for other passes in the future
-        match element_type:
-            case IntegerType():
-                bitwidth = element_type.width.data
-                if bitwidth != 32:
-                    raise DiagnosticException(
-                        f"Unsupported memref element type for riscv lowering: {element_type}"
-                    )
-                assert irdl_to_attr_constraint(
-                    DenseIntOrFPElementsAttr[IntegerType]
-                ).verifies(initial_value)
-                ints = initial_value.get_values()
-                ptr = TypedPtr.new_int32(ints).raw
-            case Float32Type():
-                assert irdl_to_attr_constraint(
-                    DenseIntOrFPElementsAttr[Float32Type]
-                ).verifies(initial_value)
-                floats = initial_value.get_values()
-                ptr = TypedPtr.new_float32(floats).raw
-            case Float64Type():
-                assert irdl_to_attr_constraint(
-                    DenseIntOrFPElementsAttr[Float64Type]
-                ).verifies(initial_value)
-                floats = initial_value.get_values()
-                ptr = TypedPtr.new_float64(floats).raw
-            case _:
-                raise DiagnosticException(
-                    f"Unsupported memref element type for riscv lowering: {element_type}"
-                )
+        if len(initial_value.data.data) % (self.xlen // 8):
+            raise DiagnosticException(
+                f"memref.global value size not divisible by xlen ({self.xlen}) not yet "
+                "supported."
+            )
 
-        text = ",".join(hex(i) for i in ptr.int32.get_list(42))
+        i32_attr = DenseArrayBase(i32, initial_value.data)
+        text = ",".join(hex(i) for i in i32_attr.iter_values())
```

Comment:
Can we not use the original `DenseIntOrFPElementsAttr`?

Reply:
This all feels a bit cursed tbh

Reply:
No, the point is that RISCV assembly expects comma-separated hex values of 32-bit integers, I could do this manually with the struct package on the original data, but this feels more convenient.